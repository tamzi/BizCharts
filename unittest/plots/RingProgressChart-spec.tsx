import React from 'react';
import RingProgressChart from '../../src/plots/RingProgressChart';
import { render, cleanup } from '@testing-library/react';


describe('Plots-RingProgressChart', () => {
  test('进度条', () => {
    let chart = null;
    render(<RingProgressChart
      width={100}
      height={100}
      percent={0.8}
      color={ ['#5B8FF9', '#E8EDF3']}
      onGetG2Instance={
        (c) => chart = c
      }
    />)
    expect(chart.options).toMatchSnapshot();
    cleanup();
  })
})
