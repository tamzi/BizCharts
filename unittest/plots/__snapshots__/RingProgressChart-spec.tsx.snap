// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plots-RingProgressChart 进度条 1`] = `
Object {
  "animation": true,
  "appendPadding": Array [
    10,
    5,
    10,
    10,
  ],
  "autoFit": true,
  "color": Array [
    "#5B8FF9",
    "#E8EDF3",
  ],
  "data": 0.8,
  "height": 100,
  "innerRadius": 0.8,
  "percent": 0.8,
  "radius": 0.98,
  "statistic": Object {
    "content": Object {
      "formatter": [Function],
      "style": Object {
        "fill": "#4D4D4D",
        "fontSize": "14px",
        "fontWeight": 300,
        "textAlign": "center",
        "textBaseline": "middle",
      },
    },
    "title": false,
  },
  "width": 100,
}
`;
